### ISSUE TITLE
winfstest fails sometimes if explorer window is open

### ISSUE THREAD

**Issue by bailey27** [dokan-dev/dokany#283]
I was editing my post, and I got some error on github.  So it was lost.  Here it is again.

I'll re-enter it
# Environment
- Windows version: Windows 10
- Processor architecture: x64
- Dokany version: various (RC3, current source from different dates)
- Library type (Dokany/FUSE): Dokany
# Check List
- [x ] I checked my issue [doesn't exist yet](https://github.com/dokan-dev/dokany/issues?utf8=%E2%9C%93&q=is%3Aissue)
- [ x] My issue is valid with mirror default sample and not specific to my user-mode driver implementation
- [x ] I can always reproduce the issue with the provided description below.
- [x ] I have updated Dokany to the latest version (https://github.com/dokan-dev/dokany/releases) and have reboot my computer after.
# Description

If I open an explorer window on M:\, and then run winfsest on m:\, I get failures on about 1 out of ten runs.

Here is an example with failures.

http://pastebin.com/0k7F6DQV

I cannot reproduce it with source from June 1.  I can from June 10.  I'm trying to narrow it down to a specific date.

It doesn't happen with RC3.  It does still happen with my unmerged fixes for oplocks for the zip issue.
# Logs

I will post logs if somebody really wants them.

**Comment by Liryna** [comment]
Hi @bailey27 ,

RC3 does have fs notifications. I don't really know what could change this behaviour since RC3.
Does that mean that more test pass with RC3 when explorer is open ? (normally some will fail since I fixed different test after RC3)

Is it always the same tests that fail ?

https://github.com/dokan-dev/dokany/compare/v1.0.0-RC3...master

**Comment by bailey27** [comment]
It seems to not have the same errors all the time.  

It seems to happen more often if I can see the winfstest files come and go in the explorer window.  On some winfstest runs I never see the files in the explorer window, and in some I do.

It seems to happen less often the more older the source.  I tried 100 winfstest runs on the June 1 source, and it did not fail..  On June 5 source it happened 1/50 runs.  I see no differences between them that should affect the outcome.

And now suddenly I cannot build anymore.  I get this error
```
"C:\git\forks\dokany\dokan.sln" (Build target) (1) ->
"C:\git\forks\dokany\sys\sys.vcxproj" (default target) (2) ->
(Inf2Cat target) ->
  C:\Program Files (x86)\Windows Kits\10\build\WindowsDriver.common.targets(125
2,5): error MSB6006: "inf2cat.exe" exited with code -2. [C:\git\forks\dokany\sy
s\sys.vcxproj]

    0 Warning(s)
    1 Error(s)
```

I tried rebooting.  It seems my computer suddenly cannot build anymore.

**Comment by marinkobabic** [comment]
If you are signing the driver your self then you will fail around the middle night. For some reason the time causes the signing to fail. You should see it somewhere in the log file.

**Comment by bailey27** [comment]
@marinkobabic ,
Thanks.  I see that now.

**Comment by bailey27** [comment]
I tried it with building from various date-based snapshots going back to June 1. June 1 consistently did not do it. (I tried 200 runs).

But I think that was a fluke.

Because I could make it fail with builds right after that which had no changes that could possibly affect this.

Also, I was able to reproduce it using the debug install of RC3 in my vm.  I am not able to reproduce it with the RC3 release install on my real machine.

I think it is because of a race condition.  

With the debug build, but with no debug prints, it happens about 1/100 or 1/50 times.

With debug prints going out (even if dbgview filter is set to like *arghh*, so it isn't printing anything).  It happens much more often.

I cannot say for sure if this is really a bug in dokany, or if having explorer causing more fs chatter during the test somehow makes winfstest fail even though dokany is working correcting.
